contrib/hashicorp/consul/consul.go

```diff
@@ -0,0 +1,162 @@
+package consul
+
+import (
+	"context"
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	consul "github.com/hashicorp/consul/api"
+)
+
+// Client wraps the regular *consul.Client and augments it with tracing. Use NewClient to initialize it.
+type Client struct {
+	*consul.Client
+
+	config *clientConfig
+	ctx    context.Context
+}
+
+// NewClient returns a traced Consul client.
+func NewClient(config *consul.Config, opts ...ClientOption) (*Client, error) {
+	c, err := consul.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+	return WrapClient(c, opts...), nil
+}
+
+// WrapClient wraps a given consul.Client with a tracer under the given service name.
+func WrapClient(c *consul.Client, opts ...ClientOption) *Client {
+	cfg := new(clientConfig)
+	defaults(cfg)
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	return &Client{c, cfg, context.Background()}
+}
+
+// WithContext sets a context on a Client. Use it to ensure that emitted spans have the correct parent.
+func (c *Client) WithContext(ctx context.Context) *Client {
+	c.ctx = ctx
+	return c
+}
+
+// A KV is used to trace requests to Consul's KV.
+type KV struct {
+	*consul.KV
+
+	config *clientConfig
+	ctx    context.Context
+}
+
+// KV returns the KV for the Client.
+func (c *Client) KV() *KV {
+	return &KV{c.Client.KV(), c.config, c.ctx}
+}
+
+func (k *KV) startSpan(resourceName string, key string) ddtrace.Span {
```

Comment:
We should rename `resourceName` to `op` and use it as the span name. The resource will automatically be made equal to it.

Reply:
I don't follow, could you point to an example in another contrib library?

Reply:
This comment was related to the changes I recommended with op name and resource name. Since we aren't doing that anymore, we can consider it resolved. It doesn't apply now.